lib/Tool/Text.php

```diff
@@ -82,7 +82,9 @@ public static function wysiwygText(?string $text, array $params = []): ?string
 
                             $site = Frontend::getSiteForDocument($element);
                             if ($site instanceof Site) {
-                                $path = Tool::getRequestScheme() . '://' . $site->getMainDomain() . preg_replace('~^' . preg_quote($site->getRootPath(), '~') . '~', '', $path);
+                                if (preg_match('~^' . preg_quote($site->getRootPath(), '~') . '~', $path)) {
```

Comment:
maybe it would be better to check if `$site->getMainDomain()` is already there? 🤔